api/v1alpha1/ocivalidator_types.go

```diff
@@ -17,17 +17,65 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"strings"
+
 	validatorv1alpha1 "github.com/spectrocloud-labs/validator/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // OciValidatorSpec defines the desired state of OciValidator
 type OciValidatorSpec struct {
-	RegistryAuthRules []RegistryAuthRule `json:"registryAuthRules,omitempty" yaml:"registryAuthRules,omitempty"`
+	OciRegistryRules []OciRegistryRule `json:"ociRegistryRules,omitempty" yaml:"ociRegistryRules,omitempty"`
+	EcrRegistryRules []EcrRegistryRule `json:"ecrRegistryRules,omitempty" yaml:"ecrRegistryRules,omitempty"`
+}
+
+func (s OciValidatorSpec) ResultCount() int {
+	return len(s.EcrRegistryRules) + len(s.OciRegistryRules)
+}
+
+type OciRegistryRule struct {
+	Host           string    `json:"host" yaml:"host"`
+	RepositoryPath string    `json:"repositoryPath,omitempty" yaml:"repositoryPath,omitempty"`
+	Auth           BasicAuth `json:"auth,omitempty" yaml:"auth,omitempty"`
+	Cert           string    `json:"cert,omitempty" yaml:"cert,omitempty"`
+}
+
+func (r OciRegistryRule) Name() string {
+	var sb strings.Builder
+	sb.WriteString(r.Host)
+	sb.WriteString("/")
+	sb.WriteString(r.RepositoryPath)
+	return sb.String()
```

Comment:
Personally, I would use `fmt.Sprintf` when constructing a string with only 3 components.

My understanding of string builders is that they're mostly suited for cases where you're constructing a string incrementally by concatenating MANY components (which may vary dynamically) and you want to avoid repeated memory reallocations as the string grows. Versus scenarios like this where the constituent elements are few and well known.